### PR TITLE
Export multiple compressed CSV with only the required columns

### PR DESCRIPTION
**Pull request by StanislasGuinel**
[CH62565] Export multiple compressed CSV to the output folder and run the LOAD CSV query on each of them
[CH37963] Export only the required columns to the output folder

### CODE REVIEW

File: python-lib/dku_neo4j/__init__.py [hunk 1]
```diff
@@ -68,21 +68,29 @@ def delete_relationships(self, params):
...
 USING PERIODIC COMMIT
-LOAD CSV FROM 'file:///{csv_file_path}' AS line FIELDTERMINATOR '\t'
+LOAD CSV FROM 'file:///{import_file_path}' AS line FIELDTERMINATOR ','
 WITH {definition}
 MERGE (n:`{params.nodes_label}` {node_primary_key_statement})
 {properties}
 """
-        logging.info(f"Neo4j plugin - Importing nodes into Neo4j: {query}")
-        self.run(query, log_results=True)
+            if i == 0:
```
Comment (alexbourret) [inline python-lib/dku_neo4j/__init__.py:88]: i )):-[

**Review by alexbourret — COMMENTED** [review]
👍 
Two points though:
- Many leftover one letter variables
- You might want to try running the plugin with a remote neo4j before settling down for a fixed batch size number

File: custom-recipes/neo4j-export-nodes/recipe.py [hunk 2]
```diff
@@ -48,8 +40,10 @@
     if params.clear_before_run:
         neo4jhandle.delete_nodes(params.nodes_label)
 
+    batch_size = 100000 if load_from_csv else 10000
```
[the remark this answers is not in the document]
Comment (StanislasGuinel) [reply custom-recipes/neo4j-export-nodes/recipe.py:43]: Waiting for the remote neo4j server on dku34 and the feedback from the neo4j guys to settle on the batch size. We also may want to make this a parameter for users

File: python-lib/dku_neo4j/__init__.py [hunk 3]
```diff
@@ -68,21 +68,29 @@ def delete_relationships(self, params):
...
         properties = self._properties(columns_list, params.node_properties, "n", params.property_names_map)
-        query = f"""
+        for i, df in enumerate(df_iterator):
+            self._check_no_empty_primary_key(df, mandatory_columns=[params.node_id_column])
+            local_path = f"dss_neo4j_export_temp_file_{i+1:03}.csv.gz"
+            import_file_path = file_handler.write(df, local_path)
+            query = f"""
 USING PERIODIC COMMIT
-LOAD CSV FROM 'file:///{csv_file_path}' AS line FIELDTERMINATOR '\t'
+LOAD CSV FROM 'file:///{import_file_path}' AS line FIELDTERMINATOR ','
```
[the remark this answers is not in the document]
Comment (StanislasGuinel) [reply python-lib/dku_neo4j/__init__.py:83]: you mean the string queries ?

File: python-lib/dku_neo4j/__init__.py (hunk 1 above)
Comment (StanislasGuinel) [reply python-lib/dku_neo4j/__init__.py:88]: ??

**Pull request merged**